images/vector/tests/main.tf

```diff
@@ -10,23 +10,29 @@ variable "digest" {
 
 data "oci_string" "ref" { input = var.digest }
 
+resource "random_id" "hex" { byte_length = 4 }
+
 resource "helm_release" "vector" {
-  name       = "vector"
-  repository = "https://helm.vector.dev"
-  chart      = "vector"
+  name             = "vector-${random_id.hex.hex}"
+  repository       = "https://helm.vector.dev"
+  chart            = "vector"
+  create_namespace = true
+  namespace        = "vector-${random_id.hex.hex}"
+  timeout          = 120
 
   values = [jsonencode({
     localpv = {
       image = {
         registry   = join("", [data.oci_string.ref.registry, "/"])
         repository = data.oci_string.ref.repo
-        tag        = data.oci_string.ref.pseudo_tag
+        tag        = trimprefix(data.oci_string.ref.digest, "sha256:")
```

Comment:
I'm sort of surprised that this works either TBQH.  Maybe it's some quirk of the helm chart that it does, but I'm guessing that this is deploying an `ImagePullBackoff` `vector` image to the k3s cluster and we just don't realize it because we don't try to use the `vector` instance while it's up.

There really isn't a `tag` that we can use here because there aren't any tags applied to the digest until tests have passed.  `pseudo_tag` is a game we play where we know that when the syntax `:foo@bar` is used, the `foo` part isn't used by anything, but it lets us tunnel a digest through tag fields in charts.

Perhaps the problem here is that `localpv` isn't actually something that appears in the `vector` Helm chart, so none of these values are actually being used 🤔 
https://github.com/vectordotdev/helm-charts/blob/00c9a2cc17cfe41a378db86d046227d24df04559/charts/vector/values.yaml#L26-L40

Reply:
I think that the `trimprefix` you have is right, if you change `tag =` to `sha =`, but my confidence that the `localpv` piece is right is close to zero (if I'm looking at the right helm chart).

Reply:
I just checked this helm installation details. localpv shouldn't have been there, unsure why I added it previously. The tests passed because they used the upstream image.